hbase_pipeline/src/main/java/com/splicemachine/derby/hbase/SpliceIndexEndpoint.java

```diff
@@ -178,6 +178,7 @@ public void stop(CoprocessorEnvironment env) throws IOException{
         if(service!=null)
             try{
                 service.shutdown();
+                DatabaseLifecycleManager.manager().deregisterGeneralService(service);
```

Comment:
If `service.shutdown()` throws an exception, deregistration will not occur. Perhaps the line above should go into a `finally` block instead, if you want deregistration to occur regardless.